### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by umeshwaghode**
### Description
This PR contains enhancements and fixes:
- related in the cx-client-common related to report enhancements to support reporting SCA Policy Violations.
-Couple of dependencies are upgraded.
-Handling for 'Improved Scan Flow' engine id
-OSA and SAST report URL has been corrected.

### References

JIRA-PLUG-432, JIRA-PLUG-485

### Testing

Tested the build through Jenkins plugin for above changes.

### Checklist

- [ X] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentaiton is a Wiki Update, please indicate desired changes within PR MD Comment*
- [X ] All active GitHub checks for tests, formatting, and security are passing
- [ X] The correct base branch is being used

### CODE REVIEW

File: src/main/resources/com/cx/report/report.ftl [hunk 1]
```diff
@@ -2685,25 +2691,44 @@
...
                     </#if>
+                   
+                    <#if (sca.policyEvaluations)??> 
+	                    <#if sca.policyEvaluations?size gt 0>
+	                        <#list sca.policyEvaluations as scaPolicy>
+	                        	<#if scaPolicy.isViolated>
+		                            <tr>
+		                                <td>${scaPolicy.name}</td>
+		                                <td id="ruleName">
+		                                	<#list scaPolicy.rules as rule>${rule.name}<#sep>,</#list>
```
Comment (Kevin-CB) [inline src/main/resources/com/cx/report/report.ftl:2722]:
```suggestion
		                                <td>${sast.encodeXSS(scaPolicy.name)}</td>
		                                <td id="ruleName">
		                                	<#list scaPolicy.rules as rule>${sast.encodeXSS(rule.name)}<#sep>,</#list>
```

Comment (Kevin-CB) [reply src/main/resources/com/cx/report/report.ftl:2722]: I guess these two also need to be escaped

**Review by Kevin-CB — APPROVED** [review]
LGTM

**Review by SubhadraSahoo — APPROVED** [review]
Looks good

**Pull request merged**